src/main/java/io/github/eocqrs/kafka/parameters/KfFlexibleSettings.java

```diff
@@ -0,0 +1,48 @@
+package io.github.eocqrs.kafka.parameters;
+
+import io.github.eocqrs.kafka.ConsumerSettings;
+import io.github.eocqrs.kafka.Params;
+import io.github.eocqrs.kafka.ParamsAttr;
+import io.github.eocqrs.kafka.ProducerSettings;
+import lombok.RequiredArgsConstructor;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * KfFlexibleSettings allow you to add custom settings.
+ *
+ * @author Ivan Ivanchuk (l3r8y@duck.com)
+ * @since 0.0.2
+ */
+@RequiredArgsConstructor
+public final class KfFlexibleSettings<K, X>
+  implements ConsumerSettings<K, X>, ProducerSettings<K, X> {
+
+  private final Params params;
+
+  @Override
+  public KafkaConsumer<K, X> consumer() {
+    return new KafkaConsumer<>(this.configuration());
+  }
+
+  @Override
+  public KafkaProducer<K, X> producer() {
+    return new KafkaProducer<>(this.configuration());
+  }
+
+  /**
+   * It takes all the parameters in the `params` object
+   * and returns a map of the parameter names and values.
+   *
+   * @return A map of the parameters and their values.
+   */
+  private Map<String, Object> configuration() {
```

Comment:
private method is evil, we need to introduce new class named `Configuration` or something like this.

Update PR with a new class or a new puzzle

Reply:
@l3r8yJ yes, each private method is the indicator that we need to instantiate a new class. The private method is not:
1) testable
2) reusable
Class is.

Reply:
@h1alexbel it is not reusable, but I assume that OOP was not introduced to reduce code duplication. And why is it not tested in cases where it is called inside the test and the test passes?